### PR TITLE
Admin supplier companies

### LINKED ISSUE #4: Allow for Supplier Companies management
Allow CRUD operations for supplier company features

### PR DESCRIPTION
**Pull request by HarryPi**
## New features
- Allow for edit suppliers
- Allow for viewing suppliers

## Linked issues
Closes #4

### CODE REVIEW

File: AdminClient/src/app/supplier/supplier.module.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {NgModule} from '@angular/core';
+
+import {SupplierRoutingModule} from './supplier-routing.module';
+import {SharedModule} from '../shared/shared.module';
+import { SupplierComponent } from './supplier.component';
+import { SupplierFormComponent } from './supplier-form/supplier-form.component';
+import {ReactiveFormsModule} from '@angular/forms';
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/supplier.module.ts:7]: Consistent styling

**Comment by HarryPi** [comment]
What is this branch? @markscamilleri  haha

**Comment by markscamilleri** [comment]
You tell me, you made it 😂

File: AdminClient/src/app/supplier/store/suppliers.actions.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { createAction, props } from '@ngrx/store';
+import { SupplierModel } from '../../shared/model/supplier/supplier.model';
+
+export const beginLoadingSuppliers = createAction(
+    '[Clients - Component] Load all supplier'
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.actions.ts:5]:
```suggestion
    '[Suppliers - Component] Load all suppliers'
```

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            {
+              width: '66vw',
+              height: '66vh',
+              panelClass: 'customerDialog'
+            }
+        );
+      })
+  ), {dispatch: false});
+
+  loadAllClients$ = createEffect(() => this.actions$.pipe(
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts]:
```suggestion
  loadAllSuppliers$ = createEffect(() => this.actions$.pipe(
```

Check references too

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      ofType(SupplierActions.beginLoadingSuppliers),
+      switchMap((action: Action) => of(this.loadAllClients()).pipe(delay(2000)))
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts:75]:
```suggestion
      switchMap((action: Action) => of(this.loadAllSuppliers()).pipe(delay(2000)))
```

Check References too

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  );
+
+  constructor(
+      private dialog: MatDialog,
+      private actions$: Actions,
+      private store$: Store<fromApp.State>
+  ) {
+  }
+
+  createDummyClients() {
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts:86]:
```suggestion
  createDummySuppliers() {
```

check references too

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        lastName: `last name ${ i }`,
+        id: `${ i }`,
+        notes: `look at me i am note ${ i } FEAR ME!`
+      });
+    }
+
+    return dummySuppliers;
+  }
+
+  loadAllClients(): Action {
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts:105]:
```suggestion
  loadAllSuppliers(): Action {
```

check references

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    const allSuppliers = this.createDummyClients();
+
+    return SupplierActions.allSuppliersLoaded({suppliers: allSuppliers});
+  };
+
+  private editCustomer(editedSupplier: SupplierModel): Action {
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts]:
```suggestion
  private editSupplier(editedSupplier: SupplierModel): Action {
```

And check references

File: AdminClient/src/app/supplier/store/suppliers.effects.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+import { PopupDialogComponent } from '../../shared/components/popup-dialog/popup-dialog.component';
+import { PopupDialogDataModel } from '../../shared/model/popup-dialog-data.model';
+import { SupplierModel } from '../../shared/model/supplier/supplier.model';
+import { SupplierFormComponent } from '../supplier-form/supplier-form.component';
+import * as SupplierActions from './suppliers.actions';
+import * as fromSuppliers from './suppliers.reducer';
+
+@Injectable()
+export class SuppliersEffects {
+  deleteCustomer$ = createEffect(() => this.actions$.pipe(
```
Comment (markscamilleri) [inline AdminClient/src/app/supplier/store/suppliers.effects.ts:17]:
```suggestion
  deleteSupplier$ = createEffect(() => this.actions$.pipe(
```

and check referneces

**Pull request merged**